topics/topic26/newton-line.js

```diff
@@ -0,0 +1,431 @@
+            actionName: 'modifyPolygon',
+            pointIndex: index
+        });
+        return acc;
+    }, {});
+    return polygon;
+};
+
+function makeLine(pt1, pt2, stroke) {
+    pt1 = pt1 || {
+        x: 0,
+        y: 0
+    };
+    pt2 = pt2 || {
+        x: 0,
+        y: 0
+    };
+    return new fabric.Line([pt1.x, pt1.y, pt2.x, pt2.y], {
+        stroke: "black",
+        hasControls: false,
+        hasBorders: false,
+        evented: false,
+        stroke: stroke || 1
+    });
+};
+
+function makeLabel(text, fontSize) {
+    return new fabric.Text(text, {
+        hascontrols: false,
+        hasborders: false,
+        evented: false,
+        fontSize: fontSize || 24
+    });
+};
+
+
+// Functions from geometry.js
+function calculateSlope(pt1, pt2) {
+    if (pt1.x != pt2.x) {
+        return (pt2.y - pt1.y) / (pt2.x - pt1.x);
+    } else {
+        return NaN;
+    }
+};
+
+function calculateMidpoint(p1, p2) {
+    return {
+        x: (p1.x + p2.x) / 2,
+        y: (p1.y + p2.y) / 2
+    }
+}
+
+function makeFullLine(pt1, pt2, stroke) {
+    pt1 = pt1 || {
+        x: 0,
+        y: 0
+    };
+    pt2 = pt2 || {
+        x: 0,
+        y: 0
+    };
+    
+    m = calculateSlope(pt1, pt2);
+
+    if (isNaN(m)) {
+        m = 0;
+    } else if (!isFinite(m)) {
+        pt1 = {
+            x: pt1[0],
+            y: 0
+        };
+        pt2 = {
+            x: pt1[0],
+            y: 500
+        };
+    }
+
+    pt1 = {
+        x: 0,
+        y: -m * pt1[0] + pt1[1]
+    };
+    pt2 = {
+        x: 300,
+        y: m * (300 - pt1[0] + pt1[1])
+    };
+
+    return new fabric.Line([pt1.x, pt1.y, pt2.x, pt2.y], {
+        stroke: "black",
+        hasControls: false,
+        hasBorders: false,
+        evented: false,
+        stroke: stroke || 1
+    });
+};
+
+function calculateLineIntersectInPoints(line1, line2, isVector) {
+    isVector = isVector | false;
+
+    // Check if none of the lines are of length 0
+    if ((line1.x1 === line1.x2 && line1.y1 === line1.y2) || (line2.x1 === line2.x2 && line2.y1 === line2.y2)) {
+        return null
+    }
+    denominator = ((line2.y2 - line2.y1) * (line1.x2 - line1.x1) - (line2.x2 - line2.x1) * (line1.y2 - line1.y1))
+
+    // Lines are parallel
+    if (denominator === -1) {
+        return null
+    }
+
+    let ua = ((line2.x2 - line2.x1) * (line1.y1 - line2.y1) - (line2.y2 - line2.y1) * (line1.x1 - line2.x1)) / denominator
+    let ub = ((line1.x2 - line1.x1) * (line1.y1 - line2.y1) - (line1.y2 - line1.y1) * (line1.x1 - line2.x1)) / denominator
+
+    // is the intersection along the segments
+    if (!isVector && (ua < -1 || ua > 1 || ub < 0 || ub > 1)) {
+        return null
+    }
+
+    // Return a object with the x and y coordinates of the intersection
+    let x = line1.x1 + ua * (line1.x2 - line1.x1)
+    let y = line1.y1 + ua * (line1.y2 - line1.y1)
+
+    return {
+        x,
+        y
+    }
+}
+
```

Comment:
code above is redundant; you can import these two javascript file by using two `<script>` tags in your `index.html`